test/data/javascript.js

```diff
@@ -104,6 +107,218 @@ exports.test_data = {
 
         ],
     }, {
+        name: "operator_position option - ensure no neswlines if preserve_newlines is false",
+        matrix: [
+            {
+                options: [
+                    { name: "operator_position", value: "'before_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'after_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'preserve_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }
+        ],
+        tests: [
+            {
+                unchanged: inputlib.operator_position.sanity
+            }, {
+                input: inputlib.operator_position.comprehensive,
+                output: inputlib.operator_position.sanity,
+            }
+        ]
+    }, {
+        name: "reset preserve_newlines and operator_position",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "operator_position", value: "'before_newline'" }
+        ]
+    }, {
+        name: "operator_position option - set to 'before_newline' (default value)",
+        tests: [
+            {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive,
+                output: [
+                    'var res = a + b - c /',
+                    '    d * e %',
+                    '    f;',
```

Comment:
Alright I'll do that - however I disagree with the alternative being synonymous with `preserve_newline` since that will preserve newlines on either side of the operator.